tests/federation_room_join_partial_state_test.go

```diff
@@ -3295,6 +3295,61 @@ func TestPartialStateJoin(t *testing.T) {
 		}
 	})
 
+	t.Run("Room stats are correctly updated once state re-sync completes", func(t *testing.T) {
+		deployment := Deploy(t, b.BlueprintAlice)
+		defer deployment.Destroy(t)
+
+		// create a user with admin powers as we will need this power to make the remote room visible in the
+		// local room list
+		terry := deployment.RegisterUser(t, "hs1", "terry", "pass", true)
+
+		server := createTestServer(t, deployment)
+		cancel := server.Listen()
+		defer cancel()
+		serverRoom := createTestRoom(t, server, terry.GetDefaultRoomVersion(t))
+
+		// start a partial state join
+		psjResult := beginPartialStateJoin(t, server, serverRoom, terry)
+		defer psjResult.Destroy(t)
+
+		// make the remote room visible in the local room list
+		reqBody := client.WithJSONBody(t, map[string]interface{}{
+			"visibility": "public",
+		})
+		terry.MustDoFunc(t, "PUT", []string{"_matrix", "client", "v3", "directory", "list", "room", serverRoom.RoomID}, reqBody)
+
+		// sanity check - before the state has completed syncing state we would expect only one user
+		// to show up in the room list
```

Comment:
Presumably terry?

I'm a little surprised by this---what's the logic/thinking here? Are we only tracking counting local users until the resync completes?

[The room is created by a user called `charlie` on the complement-controlled HS.](https://github.com/matrix-org/complement/blob/fe5887f0907b88a9c562a67009200b8b7af26545/tests/federation_room_join_partial_state_test.go#L94-L103) I would expect Charlie's membership to show up in the /send_join response: Charlie's membership should be an auth_event of the power levels event that Charlie creates, and the PL event is an auth_event of Terry's join. Therefore I'd expect Terry to be able to see Charlie (but not Derek).

Reply:
While Charlie's membership does show up in the /send_join response, it's in the auth chain only and not the current room state. Room stats gets computed off the current state only.